pdcupdater/utils.py

```diff
@@ -120,6 +124,75 @@ def ensure_release_component_exists(pdc, release_id, name):
         if body['non_field_errors'] != [message]:
             raise
 
+    # But if it was just that the component already existed, then go back and
+    # query for what we tried to submit (return the primary key)
+    query = dict(name=name, release=release_id)
+    results = pdc['release-components']._(**query)
+    if not results:
+        raise IndexError("No results found for %r after submitting %r" % (
+            query, data))
+    if len(results) > 1:
+        raise IndexError("%i results found for %r after submitting %r" % (
+            len(results), query, data))
+    return results[0]
+
+
+def ensure_release_component_relationship_exists(pdc, parent, child, type):
+    """ Create a release-component-relationship in PDC
+    only if it doesn't already exist.
+    """
+
+    ensure_release_component_exists(pdc, parent['release']['release_id'], parent['name'])
+    ensure_release_component_exists(pdc, child['release']['release_id'], child['name'])
+
+    try:
+        # Try to create it
+        pdc['release-component-relationships']._({
+            'parent': parent,
+            'child': child,
+            # This may not exist, and we have no API to create it.  It must be
+            # entered by an admin in the admin panel beforehand.
+            'type': type,
+        })
+    except beanbag.bbexcept.BeanBagException as e:
+        if e.response.status_code != 400:
+            raise
+        body = e.response.json()
+        if not 'non_field_errors' in body:
+            raise
+
+        # TODO - look for any further special psuedo-error handling cases here.
+        import pprint
+        pprint.pprint(body)
```

Comment:
I should probably change it to ``log.something(body)``.

I'd like to keep it because I'm not sure what kind of error PDC might throw back at me.  If it is a new error I've never seen before, I'd like to know the details in the logs so I can make an appropriate patch to handle it in the future.